src/test/java/seedu/eylah/diettracker/logic/commands/AddCommandTest.java

```diff
@@ -0,0 +1,218 @@
+import seedu.eylah.diettracker.model.food.Food;
+import seedu.eylah.diettracker.model.self.Height;
+import seedu.eylah.diettracker.model.self.Weight;
+import seedu.eylah.diettracker.testutil.FoodBuilder;
+
+public class AddCommandTest {
+
+    @Test
+    public void constructor_nullFood_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddCommand(null));
+    }
+
+    @Test
+    public void execute_foodAcceptedByModel_addSuccessful() throws Exception {
+        ModelStubAcceptingFoodAdded modelStub = new ModelStubAcceptingFoodAdded();
+        Food validFood = new FoodBuilder().build();
+
+        CommandResult commandResult = new AddCommand(validFood).execute(modelStub);
+
+        assertEquals(String.format(AddCommand.MESSAGE_SUCCESS, validFood), commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validFood), modelStub.foodsAdded);
+    }
+
+    //@Test
+    //public void execute_duplicateFood_throwsCommandException() {
+    //    Food validFood = new FoodBuilder().build();
+    //    AddCommand addCommand = new AddCommand(validFood);
+    //    ModelStub modelStub = new ModelStubWithFood(validFood);
+    //
+    //    assertThrows(CommandException.class, AddCommand.MESSAGE_DUPLICATE_FOOD, () -> addCommand.execute(modelStub));
+    //}
+
+    @Test
+    public void equals() {
+        Food alice = new FoodBuilder().withName("Alice").build();
+        Food bob = new FoodBuilder().withName("Bob").build();
+        AddCommand addAliceCommand = new AddCommand(alice);
+        AddCommand addBobCommand = new AddCommand(bob);
+
+        // same object -> returns true
+        assertTrue(addAliceCommand.equals(addAliceCommand));
+
+        // same values -> returns true
+        AddCommand addAliceCommandCopy = new AddCommand(alice);
+        assertTrue(addAliceCommand.equals(addAliceCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addAliceCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(addAliceCommand.equals(null));
+
+        // different food -> returns false
+        assertFalse(addAliceCommand.equals(addBobCommand));
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
+        @Override
+        public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ReadOnlyUserPrefs getUserPrefs() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public GuiSettings getGuiSettings() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setGuiSettings(GuiSettings guiSettings) {
+            throw new AssertionError("This method should not be called.");
+        }
```

Comment:
I have already created a higher-scoped Diet ModelStub class. For fixing in future you may refer to that instead.